### PR TITLE
Kool stop - accepting service arguments

### PR DESCRIPTION
**Pull request by fabriciojs**
| Issue | #87  |
| -----: | :-----: |
| :trophy: Feature | Yes |
| :warning: Break Change | No |

**Description**

Added optional arguments to `kool stop` for stopping specific containers.

---

**Notes**

The behaviour is overall preserved where we destroy the container and, if `--purge`, also remove the named volumes.

### CODE REVIEW

**Comment by danielsuguimoto** [comment]
Reading the docs, a question came to my mind: does the `rm` command really removes named volumes?

image

**Comment by fabriciojs** [comment]
@danielsuguimoto indeed, it won't erase named volumes as we use for database i.e.

I am not sure what to do. Should we warn about it? Should we error out as an incompatible combination of option and argument?

**Comment by fabriciojs** [comment]
I am lining towards simply warning the user about not removing named volumes in the case of combining the argument and --purge option. What do you think?

**Comment by danielsuguimoto** [comment]
@fabriciojs I'm ok about just warning. A note in the documentation would be nice as well. The advantage of stoping a single container is far more significant than the disadvantage of not removing the named volume, IMO. But I guess we can put a ticket for removing these volumes later in our backlog.

**Comment by fabriciojs** [comment]
@danielsuguimoto As it seems, there is not a direct way for us to remove specific volumes created by `docker-compose` on `volumes:` section without `docker-compose down -v`. But agreed we should look into it in the future. Not sure if going straight to `docker volumes` is an option, and even if so, a smart one.

**Comment by fabriciojs** [comment]
@danielsuguimoto warning added:

image

File: cmd/stop.go [hunk 1]
```diff
@@ -39,29 +40,48 @@ func NewKoolStop() *KoolStop {
...
 }
 
 // NewStopCommand initializes new kool stop command
 func NewStopCommand(stop *KoolStop) (stopCmd *cobra.Command) {
 	stopCmd = &cobra.Command{
-		Use:   "stop",
-		Short: "Stop all running containers started with 'kool start' command",
-		Args:  cobra.NoArgs,
+		Use:   "stop [SERVICE]",
+		Short: "Stop and destroy running containers started with 'kool start' command. If not SERVICE is specified as argument all containers will be stopped.",
```
Comment (danielsuguimoto) [inline cmd/stop.go:84]: I guess the correct is `If no SERVICE`, instead of not

**Review by danielsuguimoto — CHANGES_REQUESTED** [review]
Found a typo. For me, after fixing it you can merge.

**Pull request merged**